### PR TITLE
[Triton] Fix MFMA load indexing

### PR DESCRIPTION
**Pull request by binarman**
This PR fixes computation of MFMA operands loading.

### CODE REVIEW

File: python/test/unit/language/test_core_amd.py [hunk 1]
```diff
@@ -1225,6 +1225,10 @@ def kernel(X, stride_xm, stride_xn,
                                            [32, 128, 64, 2],
                                            [64, 128, 128, 2],
                                            [32, 256, 32, 8],
+                                           [32, 256, 32, 2],
+                                           [64, 32, 32, 2],
+                                           [256, 32, 32, 2],
+                                           [256, 32, 32, 4],
```
Comment (alefimov-amd) [inline python/test/unit/language/test_core_amd.py:1231]: These are regression tests, I used them for debugging.

File: lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp [hunk 2]
```diff
@@ -37,38 +37,34 @@ namespace SharedToDotOperandMFMA {
 // Computes offsets for operand A or transposed operand B
 // @param rewriter
 // @param loc
-// @param elemsPerThread operand tile shape consumed by one MFMA instruction
+// @param elemsPerInstr operand tile shape consumed by one MFMA instruction
 // @param waveM wave id for the "non K" axis
```
Comment (dfukalov) [inline lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp:41]: Seems this line should be fixed too, the new parameter is `waveId`

**Review by dfukalov — APPROVED** [review]
LGTM, but one comment fix.

File: lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp (hunk 2 above)
Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp:41]: I'll fix it when rebase it on top of github action fix.
I hope this PR will fix it: #230

Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp:41]: @jayfurmanek 
I missed when you updated this branch =(
Can I add this fix with some later PR to not rerun tests?

Comment (jayfurmanek) [reply lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp:41]: oh sorry, I should have left that to you. Adding the comment fix later is fine

**Review by jayfurmanek — APPROVED** [review]

File: lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp (hunk 2 above)
Comment (binarman) [reply lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp:41]: Fixed in https://github.com/ROCmSoftwarePlatform/triton/pull/241/files

**Pull request merged**